plugins/formal-verification/testData/diagnostics/no_contracts/extension_function.kt

```diff
@@ -3,3 +3,42 @@ fun Int.<!VIPER_TEXT!>inc<!>(): Int = this + 1
 fun <!VIPER_TEXT!>extension_method_call<!>() {
     3.inc()
 }
+
+class Foo
+fun Foo.<!VIPER_TEXT!>nothing<!>(): Unit { }
+fun Foo.<!VIPER_TEXT!>nothing2<!>(a: Int): Unit { }
+fun Foo.<!VIPER_TEXT!>inside<!>(): Unit {}
+
+class Bar() {
+    fun Foo.<!VIPER_TEXT!>inside<!>(): Unit { }
```

Comment:
My impression is that here, the function has both a `Foo` and a `Bar` `this` at its disposal.  Could you check that this actually is the case and make sure the implementation handles this correctly?

Reply:
You're right. The `this` must be discriminated using the `@` operator. The normal `this` should be dispatched to `Foo`.

Example:
```kotlin
class Bar {
    fun Foo.inside() {
        val rFoo = this@Foo
        val rBar = this@Bar
    }
}
```

Reply:
Check latest commit: https://github.com/jesyspa/kotlin/pull/133/commits/978a0947121350be1f685e31da7f816f23bc38ae